File: 卡码/代码随想录/1/06-二叉树/Js/01-二叉树遍历方式/LeetCode_0515.js
//515. 在每个树行中找最大值
import TreeNode from "../TreeNode.js";
 
const largestValues = function(root) {
  const res=new Array();
  const q=new Array();
  if(root) q.push(root);
  while(q.length>0){
    let size=q.length;
    let max=q[0].val;
    while(size-->0){
      const node=q.shift();
      max=max<node.val ? node.val : max;
      if(node.left) q.push(node.left);
      if(node.right) q.push(node.right);
    }
    res.push(max);
  }
  return res;
};

File: 卡码/代码随想录/1/06-二叉树/C++/01-二叉树遍历方式/LeetCode_0515.cpp
//515. 在每个树行中找最大值
#include <iostream>
#include <vector>
#include <queue>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  vector<int> largestValues(TreeNode* root) {
		vector<int> res;
		queue<TreeNode*> q;
		int size;
		if(root) q.push(root);
		while(!q.empty()){
			size=q.size();
			int maxVal=q.front()->val;
			while(size-->0){
				TreeNode* node=q.front();
				maxVal=node->val>maxVal ? node->val : maxVal;
				q.pop();
				if(node->left) q.push(node->left);
				if(node->right) q.push(node->right);
			}
			res.push_back(maxVal);
		}
		return res;
  }
};

int main(){
  system("pause");
  return 0;
}
